cli/README.md

```diff
@@ -0,0 +1,154 @@
+No new packages or files should be added to this base directory.
+
+[`cli/cmd/`](./cmd/) defines the root command of the CLI. Any changes concerning the root command, or additions like persistent flags (flags that can be accessed by all subcommands) should be added here.
+
+TODO: Discuss: Root command could be added to `cli/internal/cmd/`
+
+[`cli/internal/`](./internal/) holds the actual logic of the CLI, and will be discussed in detail in the following sections.
+The code is kept as internal packages to avoid accidentally exposing non public API logic of the CLI to other packages of Constellation.
+Any shared code should instead go into the global [`internal/` package](../internal/).
+
+## The internal package
+
+[`cli/internal/`](./internal/) implements logic of the CLI and is split into multiple distinct packages.
+Each package should aim to implement a specific functionality. Multi purposes packages should be avoided.
+
+The following packages are listed and explained in alphabetical order. New packages should be added here when they are created.
+
+### [`cloudcmd`](./internal/cloudcmd/)
+
+This package focuses on the interaction with the cloud provider.
+It separates the cloud provider specific code from the rest of the CLI, and
+provides a common interface for all cloud providers.
+Exported functions defined here should be cloud provider agnostic, meaning there should be no functions for specific CSPs.
+Rather the `cloudcmd` package should declare functions that take a `cloudprovier.Provider` as argument,
+perform CSP specific logic, and return a universally usable result.
+
+It is used by the [`cmd` package](#cmd) to handle creation of cloud resources and other CSP specific interactions.
+
+### [`clusterid`](./internal/clusterid)
+
+Defines the structure of the Constellation cluster ID file.
+Logic in this package should be kept minimal.
+
+### [`cmd`](./internal/cmd)
+
+Defines the commands of the CLI.
+Logic should be kept to input/output parsing whenever possible.
+Any more complex code should usually be implemented in one of the other CLI packages.
+
+The code here should be kept as cloud provider agnostic as possible.
+Any CSP specific tasks should be handled by [`cloudcmd`](#cloudcmd).
+
+All filepaths handled by the CLI code should originate from here.
+Common filepaths are defined as constants in [`internal/constants`](../internal/constants/).
+To generate workspace correct filepaths for printing, use the functions from [`workspace.go`](./internal/cmd/workspace.go) // TODO: Discuss if we should move this to its own package
+
+### [`featureset`](./internal/featureset)
+
+`featureset` defines feature gate constants for the CLI and should not implement any logic.
+
+### [`helm`](./internal/helm)
+
+Helm is used to manage Kubernetes deployments of a Constellation cluster.
+The package implements functionality to install, update, and manage Helm charts.
```

Comment:
```suggestion
The package implements functionality to install, update, and manage Helm charts.
The package includes cloud-provider specific implementations, as charts need to be rendered differently for each CSP.
Exported functions should be CSP agnostic and take a cloudprovider argument, like described for pkg `cloudcmd`.
```

Reply:
I think `helm` can be CSP specific since otherwise the `cloudcmd` pkg would have to know about helm charts which is worse.